src/lib/draw/polygon_tool.ts

```diff
@@ -246,25 +258,25 @@ export class PolygonTool {
   }
 
   // TODO Force the proper winding order that geojson requires
-  #polygonFeature() {
+  #polygonFeature(): Feature<Polygon> | null {
     if (this.points.length < 3) {
       return null;
     }
-    let f = {
+    // Deep clone here, or face the wrath of crazy bugs later!
+    let coordinates = [JSON.parse(JSON.stringify(this.points))];
+    coordinates[0].push(JSON.parse(JSON.stringify(coordinates[0][0])));
+    return {
```

Comment:
For some reason TS got confused when we did `let f = {...}; return f`, so just switched it around